packages/core/src/loadEnv.ts

```diff
@@ -34,19 +33,20 @@ export function loadEnv({
     }
   });
 
-  // clear the assigned env vars before restarting dev server
-  if (isDev()) {
-    onBeforeRestartServer(() => {
-      Object.keys(parsed).forEach((key) => {
-        if (process.env[key] === parsed[key]) {
-          delete process.env[key];
-        }
-      });
+  let _cleaned = false;
```

Comment:
Variable name do not require an underscore prefix here:

```suggestion
  let cleaned = false;
```